### PR TITLE
zh: update TOC and error-handling.md

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What is changed, added or deleted? (Required)

1. mv dm-portal to config
2. add error handling about load, similar with https://github.com/pingcap/docs-cn/pull/3655, https://github.com/pingcap/docs-cn/blob/532f7458f6668da6bfd536132550d0ec7ea708b8/load-misuse-handling.md this file will be deleted, so move the content to dm's error-handling.md

- [x] master (the latest development version, including v2.0 changes for now)
- [ ] v1.0 (TiDB DM 1.0 versions)

### CODE REVIEW

File: zh/error-handling.md [hunk 1]
```diff
@@ -148,3 +148,32 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
 在所有 DM 配置文件中，数据库相关的密码都必须使用经 dmctl 加密后的密文（若数据库密码为空，则无需加密）。有关如何使用 dmctl 加密明文密码，参见[使用 dmctl 加密上游 MySQL 用户密码](deploy-a-dm-cluster-using-ansible.md#使用-dmctl-加密上游-mysql-用户密码)。
 
 此外，在 DM 运行过程中，上下游数据库的用户必须具备相应的读写权限。在启动同步任务过程中，DM 会自动进行相应权限的前置检查，详见[上游 MySQL 实例配置前置检查](precheck.md)。
+
+### load 处理单元报错 `packet for query is too large. Try adjusting the 'max_allowed_packet' variable`
+
+#### 原因
+
+* MySQL client 和 MySQL/TiDB Server 都有 `max_allowed_packet` 配额的限制，如果在使用过程中违反其中任何一个 `max_allowed_packet` 配额，客户端程序就会收到对应的报错。目前最新版本的 Syncer、Loader、DM 和 TiDB Server 的默认 `max_allowed_packet` 配额都为 `64M`。
```
Comment (csuzhangxc) [inline zh/error-handling.md:156]: 感觉这里不需要提到 syncer、loader

File: zh/error-handling.md [hunk 2]
```diff
@@ -148,3 +148,32 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+    依据默认的 `--statement-size` 设置，DM 的 dump 处理单元默认生成的 `Insert Statement` 大小会尽量接近在 `1M` 左右，使用默认值就可以确保绝大部分情况不会出现该问题。
+
+    有时候在 dump 过程中会出现下面的 `WARN` log，但是这个报错不影响 dump 的过程，只是表达了 dump 的表可能是宽表。
+
+    ```
+    Row bigger than statement_size for xxx
+    ```
+
+* 如果宽表的单行超过了 `64M`，那么需要修改以下配置，并且使之生效。
+    * 在 TiDB Server 执行 `set @@global.max_allowed_packet=134217728` （`134217728 = 128M`）
```
Comment (csuzhangxc) [inline zh/error-handling.md:179]: DM task 配置也要改吗？

Comment (WangXiangUSTC) [reply zh/error-handling.md:179]: 嗯，也需要改的，写的时候看loader 项没有这个配置所以没写，实际上可以改 target-database 的配置

File: zh/error-handling.md (hunk 1 above)
Comment (WangXiangUSTC) [reply zh/error-handling.md]: 去掉了

File: zh/error-handling.md [hunk 3]
```diff
@@ -175,5 +175,6 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
     Row bigger than statement_size for xxx
     ```
 
-* 如果宽表的单行超过了 `64M`，那么需要修改以下配置，并且使之生效。
+* 如果宽表的单行超过了 `64M`，那么需要修改以下两项配置，并且使之生效。
     * 在 TiDB Server 执行 `set @@global.max_allowed_packet=134217728` （`134217728 = 128M`）
+    * 根据实际情况为 DM 的任务配置文件中的 `target-database` 增加配置 `max-allowed-packet: 134217728`(128M)，然后 stop 任务后再重新 start 任务。
```
Comment (csuzhangxc) [inline zh/error-handling.md:182]:
```suggestion
    * 根据实际情况为 DM 的任务配置文件中的 `target-database` 增加配置 `max-allowed-packet: 134217728`(128M)，然后 `stop-task` 后再重新 `start-task`。
```

File: zh/error-handling.md [hunk 4]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+
+    * 工作量大
+
+    * 复杂度高，不容易保证正确性
+
+    * 性能的极大降低
+
+#### 解决方案
+
+* 依据上面的原因，在代码层面不能简单的解决这个困扰，我们推荐的方式是：在 DM dump 处理单元提供的配置 `extra-args` 中设置 `statement-size`:
```
Comment (toutdesuite) [inline zh/error-handling.md:168]:
```suggestion
* 推荐在 DM 的 dump 处理单元提供的配置 `extra-args` 中设置 `statement-size`:
```

File: zh/error-handling.md [hunk 5]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+
+    依据默认的 `--statement-size` 设置，DM 的 dump 处理单元默认生成的 `Insert Statement` 大小会尽量接近在 `1M` 左右，使用默认值就可以确保绝大部分情况不会出现该问题。
+
+    有时候在 dump 过程中会出现下面的 `WARN` log，但是这个报错不影响 dump 的过程，只是表达了 dump 的表可能是宽表。
+
+    ```
+    Row bigger than statement_size for xxx
+    ```
+
+* 如果宽表的单行超过了 `64M`，那么需要修改以下两项配置，并且使之生效。
```
Comment (toutdesuite) [inline zh/error-handling.md:178]:
```suggestion
* 如果宽表的单行超过了 `64M`，需要修改以下两项配置，并且确保其生效。
```

File: zh/error-handling.md [hunk 6]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+
+    * 在 TiDB Server 执行 `set @@global.max_allowed_packet=134217728` （`134217728 = 128M`）
+
+    * 根据实际情况为 DM 的任务配置文件中的 `target-database` 增加配置 `max-allowed-packet: 134217728`(128M)，然后 `stop-task` 后再重新 `start-task`。
```
Comment (toutdesuite) [inline zh/error-handling.md:182]:
```suggestion
    * 根据实际情况为 DM 的任务配置文件中的 `target-database` 增加配置 `max-allowed-packet: 134217728`(128M)，执行 `stop-task` 后再重新 `start-task`。
```

File: zh/error-handling.md [hunk 7]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+
+    * 复杂度高，不容易保证正确性
+
+    * 性能的极大降低
+
+#### 解决方案
+
+* 依据上面的原因，在代码层面不能简单的解决这个困扰，我们推荐的方式是：在 DM dump 处理单元提供的配置 `extra-args` 中设置 `statement-size`:
+
+    依据默认的 `--statement-size` 设置，DM 的 dump 处理单元默认生成的 `Insert Statement` 大小会尽量接近在 `1M` 左右，使用默认值就可以确保绝大部分情况不会出现该问题。
```
Comment (toutdesuite) [inline zh/error-handling.md:170]:
```suggestion
    依据默认的 `--statement-size` 设置，DM 的 dump 处理单元默认生成的 `Insert Statement` 大小一般会在 `1M` 左右，使用默认值就可以确保绝大部分情况 load 处理单元不会报错 `packet for query is too large. Try adjusting the 'max_allowed_packet' variable`。
```

File: zh/error-handling.md [hunk 8]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
+
+    有时候在 dump 过程中会出现下面的 `WARN` log，但是这个报错不影响 dump 的过程，只是表达了 dump 的表可能是宽表。
```
Comment (toutdesuite) [inline zh/error-handling.md:172]:
```suggestion
    有时候在 dump 过程中会出现下面的 `WARN` log。这个报错不影响 dump 的过程，只是说明 dump 的表可能是宽表。
```

File: zh/error-handling.md [hunk 9]
```diff
@@ -148,3 +148,35 @@ aliases: ['/docs-cn/tidb-data-migration/dev/troubleshoot-dm/', '/docs-cn/tidb-da
...
 
 此外，在 DM 运行过程中，上下游数据库的用户必须具备相应的读写权限。在启动同步任务过程中，DM 会自动进行相应权限的前置检查，详见[上游 MySQL 实例配置前置检查](precheck.md)。
+
+### load 处理单元报错 `packet for query is too large. Try adjusting the 'max_allowed_packet' variable`
+
+#### 原因
+
+* MySQL client 和 MySQL/TiDB Server 都有 `max_allowed_packet` 配额的限制，如果在使用过程中违反其中任何一个 `max_allowed_packet` 配额，客户端程序就会收到对应的报错。目前最新版本的 DM 和 TiDB Server 的默认 `max_allowed_packet` 配额都为 `64M`。
+
+* DM 的全量数据导入处理模块不支持对 dump sqls 文件进行切分，原因是 DM 的 dump 处理单元采用了最简单的编码实现，，如果在 DM 实现文件切分，那么需要在 `TiDB parser` 基础上实现一个完备的解析器才能正确的处理数据切分，但是随之会带来以下的问题：
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply zh/error-handling.md:158]: dump sqls 是指 dump 处理单元导出的 sql 文件，要不改成这样：
```
DM 的全量数据导入处理模块不支持对 dump 处理模块导出的 SQL 文件进行切分。因为 DM 的 dump 处理单元采用了最简单的编码实现，如果在 DM 实现文件切分，需要在 `TiDB Parser` 基础上实现一个完备的解析器才能正确的处理数据切分。但是随之会带来以下的问题：
```

[the remark this answers is not in the document]
Comment (toutdesuite) [reply zh/error-handling.md]: updated~

**Pull request merged**